src/components/DeploymentForm.vue

```diff
@@ -119,14 +134,23 @@
   const { value: workPoolName } = useField<string | null>('workPoolName')
   const { value: workQueueName } = useField<string | null>('workQueueName')
   const { value: tags } = useField<string[] | null>('tags')
+  const { value: infrastructureOverrides } = useField<string>('infrastructureOverrides')
 
   const emit = defineEmits<{
     (event: 'submit', value: DeploymentUpdate): void,
     (event: 'cancel'): void,
   }>()
 
   const submit = handleSubmit((values) => {
-    emit('submit', values)
+    try {
+      const deploymentUpdate: DeploymentUpdate = {
+        ...values,
+        infrastructureOverrides: JSON.parse(infrastructureOverrides.value),
+      }
+      emit('submit', deploymentUpdate)
+    } catch (error) {
+      showToast(localization.error.invalidJSON, 'error')
+    }
```

Comment:
I think we're still missing the `isJson` validation rule for the overrides. Once that's added then I think this try catch can be removed. You won't be able to get here without the json being valid.